tests/src/main/java/com/hazelcast/stabilizer/tests/map/PredicatesTest.java

```diff
@@ -0,0 +1,351 @@
+package com.hazelcast.stabilizer.tests.map;
+
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.core.IMap;
+import com.hazelcast.logging.ILogger;
+import com.hazelcast.logging.Logger;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.*;
+import com.hazelcast.query.Predicate;
+import com.hazelcast.query.SqlPredicate;
+import com.hazelcast.stabilizer.probes.probes.IntervalProbe;
+import com.hazelcast.stabilizer.test.TestContext;
+import com.hazelcast.stabilizer.test.TestRunner;
+import com.hazelcast.stabilizer.test.annotations.*;
+import com.hazelcast.stabilizer.tests.helpers.StringUtils;
+import com.hazelcast.stabilizer.test.utils.ThreadSpawner;
+import com.hazelcast.stabilizer.worker.Metronome;
+import com.hazelcast.stabilizer.worker.SimpleMetronome;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class PredicatesTest {
+
+    public static enum SerializationStrategy {
+        DATA_SERIALIZABLE,
+        JAVA_SERIALIZABLE,
+        IDENTIFIED_DATA_SERIALIZABLE,
+        PORTABLE,
+    }
+
+    private final static ILogger log = Logger.getLogger(PredicatesTest.class);
+    private final AtomicLong operations = new AtomicLong();
+    // properties
+    public int threadCount = 40;
+    public int keyLength = 10;
+    // number of keys per member
+    public int keyCount = 10000;
+    public int logFrequency = 10000;
+    public int performanceUpdateFrequency = 1;
+    public static final String[] names = {"aaa", "bbb", "ccc", "ddd", "eee", "fff", "ggg"};
```

Comment:
Old version of this test has it kind of base class but this situation causes some fake serialization issue like this http://stackoverflow.com/questions/18417737/java-serialization-for-extended-class